script/clang-format-check.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+res=0
+
+clang_format_binary="$1"
+
+shift
+
+for v in $@; do
+    comp_res=$("$clang_format_binary" -style=file "$v" | diff "$v" -)
+    if [ ! -z "$comp_res" ] ; then
+        echo "$v failed the clang-format conformance check." >&2
+        res=1
+    fi
+done
+
+if [ $res -eq 0 ]; then
+    echo "Succeeded checking conformance to clang-format."
+else
+    echo "Failed checking conformance to clang-format." >&2
```

Comment:
Can we add the command which the user must run to autoformat the code to fix the error.

Reply:
This PR also adds a `clang-format` target which will do it for you.

Reply:
But you need clang 8 for that, right? If so we should mention it in the error message.

Reply:
I will extend the error message.